jobs/RAG/rag.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2023-2024, Inclusive Design Institute
+#
+# Licensed under the BSD 3-Clause License. You may not use this file except
+# in compliance with this License.
+#
+# You may obtain a copy of the BSD 3-Clause License at
+# https://github.com/inclusive-design/baby-bliss-bot/blob/main/LICENSE
+
+import sys
+import os
+from langchain_community.document_loaders import TextLoader
+from langchain_text_splitters import CharacterTextSplitter
+from langchain_community.vectorstores import FAISS
+from langchain_community.embeddings import HuggingFaceEmbeddings
+from langchain_community.chat_models import ChatOllama
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.prompts import ChatPromptTemplate
+
+
+# A utility function that prints the script usage then exit
+def printUsageThenExit():
+    print("Usage: python rag.py <sentence_transformer_model_directory>")
+    sys.exit()
+
+
+# Read the path to the sentence transformer model
+if len(sys.argv) != 2:
+    printUsageThenExit()
+else:
+    sentence_transformer_dir = sys.argv[1]
+    if not os.path.isdir(sentence_transformer_dir):
+        printUsageThenExit()
+
+# The location of the user document
+user_doc = "./data/user_doc.txt"
+
+loader = TextLoader(user_doc)
+documents = loader.load()
+# print(f"Loaded documents (first 2 rows):\n{documents[:2]}\n\n")
```

Comment:
As above in `chat_history_with_summary.py`, another commented out `print()` -- for debugging?